agreementbot/agreementbot.go

```diff
@@ -387,6 +400,44 @@ func (w *AgreementBotWorker) InitiateAgreementProtocolHandler(protocol string) {
 						work <- agreementWork
 						glog.V(5).Infof("AgreementBot queued agreement cancellation")
 
+					case *BlockchainEventCommand:
+						glog.V(5).Infof("AgreementBot received blockchain event.")
+						cmd, _ := command.(*BlockchainEventCommand)
+
+						// Unmarshal the raw event
+						if rawEvent, err := protocolHandler.DemarshalEvent(cmd.Msg.RawEvent()); err != nil {
+							glog.Errorf("AgreementBotWorker unable to demarshal raw event %v, error: %v", cmd.Msg.RawEvent(), err)
+						} else if !protocolHandler.AgreementCreated(rawEvent) && !protocolHandler.ProducerTermination(rawEvent) && !protocolHandler.ConsumerTermination(rawEvent) {
```

Comment:
I think you can omit the check for !ConsumerTermination() because you're going to ignore that event anyway.